operator/internal/handlers/lokistack_create_or_update.go

```diff
@@ -204,6 +204,20 @@ func CreateOrUpdateLokiStack(
 		}
 	}
 
+	proxyCfg, err := manifests.GetProxyConfig(k.Client)
```

Comment:
Requests for active resources on the k8s apiserver should be place in the `manifests` package. The `manifests` package is dedicated for creating resource manifests (e.g. deployments, services, configmaps, etc.) from static inputs passed via `manifests.Options`. Consider moving the code using the k8s client to an internal package dedicated for the `handlers` packages, e.g. rules, gateway secrets in https://github.com/grafana/loki/tree/main/operator/internal/handlers/internal

From the resulting configmap we extract what we need and pass it as raw inputs into a new `manifests.Options` field. This inputs can be used to create the trusted ca bundle configmap statically.